src/notes/notes.service.ts

```diff
@@ -123,65 +177,163 @@ export class NotesService {
     return note;
   }
 
-  async deleteNoteByIdOrAlias(noteIdOrAlias: string) {
-    const note = await this.getNoteByIdOrAlias(noteIdOrAlias);
+  /**
+   * @async
+   * Delete a note
+   * @param {Note} note - the note to delete
+   * @return {Note} the note, that was deleted
+   * @throws {NotInDBError} there is no note with this id or alias
+   */
+  async deleteNote(note: Note): Promise<Note> {
     return await this.noteRepository.remove(note);
   }
 
-  async updateNoteByIdOrAlias(
-    noteIdOrAlias: string,
-    noteContent: string,
-  ): Promise<Note> {
-    const note = await this.getNoteByIdOrAlias(noteIdOrAlias);
+  /**
+   * @async
+   * Update a notes content.
+   * @param {Note} note - the note
+   * @param {string} noteContent - the new content
+   * @return {Note} the note with a new revision and new content
+   * @throws {NotInDBError} there is no note with this id or alias
+   */
+  async updateNote(note: Note, noteContent: string): Promise<Note> {
     const revisions = await note.revisions;
     //TODO: Calculate patch
     revisions.push(Revision.create(noteContent, noteContent));
     note.revisions = Promise.resolve(revisions);
     return this.noteRepository.save(note);
   }
 
-  updateNotePermissions(
-    noteIdOrAlias: string,
+  /**
+   * @async
+   * Update a notes permissions.
+   * @param {Note} note - the note
+   * @param {NotePermissionsUpdateDto} newPermissions - the permissions the not should be set to
+   * @return {Note} the note with the new permissions
+   * @throws {NotInDBError} there is no note with this id or alias
+   * @throws {PermissionsUpdateInconsistentError} the new permissions specify a user or group twice.
+   */
+  async updateNotePermissions(
```

Comment:
Should it only be possible to delete permissions at once?

Reply:
this should be fixed now